### ISSUE TITLE
objective argument

### ISSUE THREAD

**Issue by killerT1992** [dmlc/xgboost#3405]
A standard objective function is defined by obj = loss + complexity. For my task I can define the loss and its derivatives (`grad` and `hess`) as needed for the `objective` argument of `XGBRegressor()`, but will this method take care of the **complexity's derivatives** as defined [here](https://xgboost.readthedocs.io/en/latest/model.html)? In other words, I want to leave the complexity penalty the same - will the method still train with a complexity penalty applied even if i don't specify it in my `objective` function?

Apologies if this isn't the correct place to post such a question, but I have tried asking on SO and no one has answered. Would really appreciate an answer here. Thank you.

**Comment by henrygouk** [comment]
Yes. The complexity penalties are applied elsewhere, so you don't need to incorporate them into the objective function.
